platform/drm/cog-platform-drm.c

```diff
@@ -669,7 +669,7 @@ init_egl (void)
     if (s_eglGetPlatformDisplay)
         egl_data.display = s_eglGetPlatformDisplay (EGL_PLATFORM_GBM_KHR, gbm_data.device, NULL);
     else
-        egl_data.display = eglGetDisplay((NativeDisplayType *) gbm_data.device);
```

Comment:
Definitively there is something weird there.

However since `typedef EGLNativeDisplayType NativeDisplayType;` (ref: https://registry.khronos.org/EGL/api/EGL/eglplatform.h) and `EGLDisplay eglGetDisplay(NativeDisplayType native_display);` (ref: https://registry.khronos.org/EGL/sdk/docs/man/html/eglGetDisplay.xhtml)

What if the change is simply `egl_data.display = eglGetDisplay((NativeDisplayType) gbm_data.device);` ?

Reply:
Would we change it also here

https://github.com/Igalia/cog/blob/master/platform/wayland/cog-platform-wl.c#L2136 ??

Reply:
Works for me.